stdlib/public/core/StringCharacterView.swift

```diff
@@ -57,17 +57,34 @@ extension String: BidirectionalCollection {
   ///   `endIndex`.
   /// - Returns: The index value immediately after `i`.
   public func index(after i: Index) -> Index {
+    let i = _guts.ensureMatchingEncoding(i)
```

Comment:
Oh cool, this isn't already inlinable :-). Would we want to consider inlining the matching encoding check so that a series of API calls can coalesce them?

Reply:
Yep, that would probably be the logical choice if this ends up being a significant performance regression.

I’m hoping we won’t need to do that, though, as we’d have to figure out how to add an inlinable wrapper around this. Playing tricks with SILGen names is the route we’ve chosen recently, but I wonder if we’d better come up with a less invasive alternative.